plugins/by-name/clangd-extensions/default.nix

```
@@ -1,271 +1,161 @@
       };
 
-      highlights = {
-        detail = helpers.defaultNullOpts.mkStr "Comment" "";
+      symbol_info = {
+        border = borderOpt;
       };
     };
 
-    memoryUsage = {
-      border = borderOpt;
-    };
-
-    symbolInfo = {
-      border = borderOpt;
-    };
+  extraOptions = {
+    enableOffsetEncodingWorkaround = lib.mkEnableOption ''
+      UTF-16 offset encoding. This is used to work around the warning:
+      "multiple different client offset_encodings detected for buffer, this is
+      not supported yet".
+    '';
   };
 
-  config =
-    let
-      setupOptions =
-        with cfg;
-        {
-          inlay_hints = with inlayHints; {
-            inherit inline;
-            only_current_line = onlyCurrentLine;
-            only_current_line_autocmd = onlyCurrentLineAutocmd;
-            show_parameter_hints = showParameterHints;
-            parameter_hints_prefix = parameterHintsPrefix;
-            other_hints_prefix = otherHintsPrefix;
-            max_len_align = maxLenAlign;
-            max_len_align_padding = maxLenAlignPadding;
-            right_align = rightAlign;
-            right_align_padding = rightAlignPadding;
-            inherit highlight priority;
-          };
-          ast = with ast; {
-            role_icons = with roleIcons; {
-              inherit
-                type
-                declaration
-                expression
-                statement
-                specifier
-                ;
-              "template argument" = templateArgument;
-            };
-            kind_icons = with kindIcons; {
-              Compound = compound;
-              Recovery = recovery;
-              TranslationUnit = translationUnit;
-              PackExpansion = packExpansion;
-              TemplateTypeParm = templateTypeParm;
-              TemplateTemplateParm = templateTemplateParm;
-              TemplateParamObject = templateParamObject;
-            };
-            highlights = with highlights; {
-              inherit detail;
-            };
-          };
-          memory_usage = with memoryUsage; {
-            inherit border;
-          };
-          symbol_info = with symbolInfo; {
-            inherit border;
-          };
-        }
-        // cfg.extraOptions;
-    in
-    mkIf cfg.enable {
-      warnings = optional (!config.plugins.lsp.enable) ''
-        You have enabled `clangd-extensions` but not the lsp (`plugins.lsp`).
-        You should set `plugins.lsp.enable = true` to make use of the clangd-extensions' features.
-      '';
-
-      plugins.lsp = {
-        servers.clangd = {
-          # Enable the clangd language server
-          enable = true;
+  extraConfig = cfg: {
+    warnings = lib.optionals (!config.plugins.lsp.enable) ''
+      Nixvim (plugins.clangd-extensions): You have enabled `clangd-extensions` but not the lsp (`plugins.lsp`). You
+      should set `plugins.lsp.enable = true` to make use of the
+      clangd-extensions' features.
```

```suggestion
      Nixvim (plugins.clangd-extensions): You have enabled `clangd-extensions` but not the lsp (`plugins.lsp`).
      You should set `plugins.lsp.enable = true` to make use of the clangd-extensions' features.
```

Unlike option descriptions which are markdown, warnings are printed as-is to the console. So we need to ensure the lines are wrapped as intended, rather than wrapping them to fit nicely in our source code.